frontend/src/Utils/api-discovery.types.ts

```diff
@@ -0,0 +1,79 @@
+import { K8sModelCommon } from '@openshift/dynamic-plugin-sdk-utils';
+import { Action, Store } from 'redux';
+
+// eslint-disable-next-line no-shadow
+export declare enum ActionType {
+  ReceivedResources = 'resources',
+  SetResourcesInFlight = 'setResourcesInFlight',
+  SetBatchesInFlight = 'setBatchesInFlight',
+  StartWatchK8sObject = 'startWatchK8sObject',
+  StartWatchK8sList = 'startWatchK8sList',
+  ModifyObject = 'modifyObject',
+  StopWatchK8s = 'stopWatchK8s',
+
+  Errored = 'errored',
+  Loaded = 'loaded',
+  BulkAddToList = 'bulkAddToList',
+  UpdateListFromWS = 'updateListFromWS',
+  FilterList = 'filterList',
+}
+
+export type K8sResourceKindReference = string;
+
+export type InitAPIDiscovery = (store: Store<unknown, Action<string>>, preferenceList?: string[]) => void;
+
+export type DiscoveryResources = {
```

Comment:
@karelhala it looks like this type should be exported explicitly?

See: https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/lib-utils/src/index.ts#L32